casper/src/main/resources/PoS.rhox

```diff
@@ -366,104 +507,97 @@ in {
-                                          for (@validatorsToWithdrawList <- validatorsToWithdrawListCh) {
-                                            @ListOps!("map", validatorsToWithdrawList, *computePay, *paymentsCh) |
-                                            contract computePay(@pk, resultCh) = {
-                                              resultCh!((pk, allBonds.get(pk) + rewards.getOrElse(pk, 0)))
-                                            } |
-                                            match validatorsToWithdrawList.toSet() {
-                                              validatorsToWithdrawSet => {
-                                                @ListOps!("filter", allBonds.toList(), *notWithdrawn, *newBondsListCh) |
-                                                @ListOps!("filter", withdrawers.toList(), *notWithdrawn, *newWithdrawersListCh) |
-                                                @ListOps!("filter", rewards.toList(), *notWithdrawn, *newRewardsListCh) |
-                                                for (@newWithdrawersList <- newWithdrawersListCh;
-                                                    @newBondsList       <- newBondsListCh;
-                                                    @newRewardsList     <- newRewardsListCh;
-                                                    @payments           <- paymentsCh) {
-                                                  returnCh!((
-                                                    payments,
-                                                    newBondsList.toMap(),
-                                                    newWithdrawersList.toMap(),
-                                                    // newRewardsList.toMap()
-                                                  ))
-                                                } |
-                                                // Checks that the validator is not a withdrawer.
-                                                contract notWithdrawn(@(pk, _), resultCh) = {
-                                                  resultCh!(not validatorsToWithdrawSet.contains(pk))
-                                                }
-                                              }
-                                            }
+                                    @ListOps!("filter", state.get("withdrawers").toList(), *isQuarantineFinished, *quarantinedValidatorsCh) |
+                                    for (@quarantinedValidators <- quarantinedValidatorsCh) {
+                                      @ListOps!("fold", quarantinedValidators, state, *computeRemove, *returnCh) |
+                                      contract computeRemove(@(pk, (bonds, quantine)), @updatedState, resultCh) = {
+                                        new payReturnCh in {
+                                          payWithdrawer!((pk, bonds + updatedState.get("comittedRewards").getOrElse(pk, 0)), *payReturnCh) |
```

Comment:
We can skip transfers if amount is 0, in case of slashed validator.

Reply:
slashed validator won't be withdrawers. Why slashed validators have to be a withdrawers?

Reply:
I agree with this change. Later we can adapt to partial slashing where some amount of stake will be left.